src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1297,7 +1297,7 @@ void EmitMultiChar(RegexNode node, bool emitLengthCheck = true)
                     bool useMultiCharReads = !caseInsensitive && byteStr.Length >= sizeof(uint);
                     if (useMultiCharReads)
                     {
-                        writer.WriteLine($"byteStr = ref global::System.Runtime.InteropServices.MemoryMarshal.GetReference(global::System.Runtime.InteropServices.MemoryMarshal.AsBytes({textSpanLocal}));");
+                        writer.WriteLine($"byteSpan = global::System.Runtime.InteropServices.MemoryMarshal.AsBytes({textSpanLocal});");
```

Comment:
Currently, strings are limited to 1bn characters (2bn bytes). Is it possible that the maximum string size might get bumped to 2bn chars, just like how szarrays can hold 2bn elements? If so, `MemoryMarshal.AsBytes(string)` could fail at runtime.